truenas/webui
# Simplify HA upgrade logic + NAS-131325: `system.reboot.info` and `failover.reboot.info` methods and events

**Changes:**

This PR contains changes for 2 tickets

1. [NAS-131325](https://ixsystems.atlassian.net/browse/NAS-131325):`system.reboot.info` and `failover.reboot.info` methods and events

   here's what's implemented for this ticket:
   1. Calling new endpoints `system.reboot.info` and `failover.reboot.info`  based on whether system is HA licensed,
   2. Showing indicator on the top bar
   3. Opening a dialog on login

2. [NAS-131326](https://ixsystems.atlassian.net/browse/NAS-131326): Simplify HA upgrade logic

   TODO

**Testing:**

1. [NAS-131325](https://ixsystems.atlassian.net/browse/NAS-131325):`system.reboot.info` and `failover.reboot.info` methods and events

   

2. [NAS-131326](https://ixsystems.atlassian.net/browse/NAS-131326): Simplify HA upgrade logic

   To test the `failover.upgrade` job, follow these steps:

   1. Login into the HA-enabled server, for example at 
   2. Go to **System > Update**
   4. Choose Manual Update , reject a prompt to make a backup, and choose the **.update** file
   5. In a new browser tab go to **Dashboard** page (or any other page)

     Now you can see the **Update in Progress** notification:
      
   ![image](https://github.com/user-attachments/assets/d493366b-ab4b-468e-90f0-09039bef3a42)


### Downstream


|Affects         |Reasoning
|----------------|-------------------------------
|Documentation   |
